docs/source/Plugin/P003_commands.repl

```diff
@@ -23,11 +23,13 @@
         N.B. The set value is the internal value, so any present formula will be processed after this value is set.
         "
         "
-        ``LogPulseStatistic,<taskindex>``
+        ``LogPulseStatistic,<cmd>,<taskindex>``
```

Comment:
You can also use the relatively new way of addressing a command to a specific task.
See [Documentation - Command a specific task for multiple instances of a plugin](https://espeasy.readthedocs.io/en/latest/Reference/Command.html?#command-a-specific-task-for-multiple-instances-of-a-plugin)

Reply:
Hmm, looks good, and better than the current solution, but does not only need changing the tested new code, but also the old code and updating the documentation again. :-(. Idea comes a little late.
For now, I would only add a FixIt commend into the code for later enhancement.

Reply:
The advantage of fixing it now is that you don't need to implement "backwards compatibility".
It is really frustrating that things you implement will get used by users ;)
If you change stuff used by others, you will render their installation not working anymore.

Reply:
Right, and that is exactly the point, because I am reluctant to change it, because the existing (old) commands
```
ResetPulseCounter,<taskIndex>
SetPulseCounterTotal,value,<taskIndex>
```
use the old way to address the taskIndex.

Would it be a good idea to use now only for the new 
`LogPulseStatistic,<cmd>,<taskindex>`
the new method
`"LogPulseStatistic,<cmd>" `
where the taskIndex is not visible any more?

What do you think?

Reply:
Well if for the same plugin we already use other commands with a taskindex as parameter, then leave it like this.

Reply:
And what if you make the taskIndex part optional, ``ResetPulseCounter[,<taskIndex>]``? When not supplied, it should use the 'current' taskIndex, thus also accommodating the ``[taskName].`` prefix.

Reply:
'Current' can also be interpreted as "first" so having it optional is a good idea anyway.

Reply:
@Ton: Good and also compatible idea.
However I'd like to leave it for the next revision, because I just had completed development and pushed the result including the documentation.
It will take me again too much time rolling that all through again.
Later .... OK

Reply:
Ok, since I needed to change the code again, I will do a go for it.
Hope solution comes soon as Easter present.

Reply:
Ok, done.
It was more work than expected, because I revised the command handling. I needed to change the new command, because of an unexpected behaviour of parseString(...). C.f.  [](https://www.letscontrolit.com/forum/viewtopic.php?f=6&t=8453&p=51617&hilit=feature#p51617)
The subcommand options of "LogPulseStatistic" is now reduced and I omitted here the optional TaskNumber parameter, because two optional parameters were nor possible because of the above bug.
So, the plugin now still supports for the existing old commands the legacy taskNumber addressing and also the new with prefix. The new "LogPulseStatistic" only supports the prefix method.
The documentation P003_commands.repl is updated.
I will publish the new version now.

Hint: When I created the documentation with ".\make.bat html" (Sphinx) I got a couple of errors from other documentation parts, that I didn't touch.